File: TerminBA/TerminBA.UI/terminba_mobile/lib/model/cancelation_notification_response.dart
import 'package:json_annotation/json_annotation.dart';

import 'package:terminba_mobile/model/reservation_response.dart';

part 'cancelation_notification_response.g.dart';

@JsonSerializable(explicitToJson: true)
class CancelationNotificationResponse {
  final int id;
  final int postOwnerId;
  final int reservationId;
  final String requesterName;
  final String facilityName;
  final String dateCancelled;
  final bool isSeen;
  final ReservationResponse? reservation;

  const CancelationNotificationResponse({
    required this.id,
    required this.postOwnerId,
    required this.reservationId,
    required this.requesterName,
    required this.facilityName,
    required this.dateCancelled,
    required this.isSeen,
    this.reservation,
  });

  factory CancelationNotificationResponse.fromJson(Map<String, dynamic> json) =>
      _$CancelationNotificationResponseFromJson(json);

  Map<String, dynamic> toJson() => _$CancelationNotificationResponseToJson(this);
}

File: TerminBA/TerminBA.Models/Model/CancelationNotificationResponse.cs
using System;

namespace TerminBA.Models.Model
{
    public class CancelationNotificationResponse
    {
        public int Id { get; set; }
        public int PostOwnerId { get; set; }
        public int ReservationId { get; set; }
        public string RequesterName { get; set; } = string.Empty;
        public string FacilityName { get; set; } = string.Empty;
        public DateTime DateCancelled { get; set; }
        public bool IsSeen { get; set; }
        public ReservationResponse? Reservation { get; set; }
    }
}
